### PR TITLE
[improvement](profile)fix profile may cause query slow

### LINKED ISSUE #11385: [Enhancement] Fix open query profile may cause query slow
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

write query profile(set enable_profile=true) may task much time, this operation should not block return query result to mysql client.
This can cause query slower than ```set enable_profile = false```.

### Solution

WriteProfile after return result to mysql client.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wangbo**
# Proposed changes

Issue Number: close #11385

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by luozenglin** [comment]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java [hunk 1]
```diff
@@ -420,6 +420,12 @@ private void finalizeCommand() throws IOException {
 
         MysqlChannel channel = ctx.getMysqlChannel();
         channel.sendAndFlush(packet);
+        // note(wb) we should write profile after return result to mysql client
+        // because write profile maybe take too much time
+        // explain query stmt do not have profile
+        if (executor != null && !executor.getParsedStmt().isExplain()) {
+            executor.writeProfile(true);
```
Comment (stalary) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:427]: Whether instanceof queryStmt judged before, do you want to expand the profile scope?

Comment (wangbo) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:427]: I want to write profile after query result return to client.

Comment (stalary) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:427]: > I want to write profile after query result return to client.
I know the idea, but I mean the side effect of this move is to expand the scope of the profile. Such as insert, use etc. also write profile.

Comment (stalary) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:427]: > I want to write profile after query result return to client.

I know the idea, but I mean the side effect of this move is to expand the scope of the profile. Such as insert, use etc. also write profile.

Comment (wangbo) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:427]: It seems that it didn't expand the scope of the profile

Comment (wangbo) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:427]: Do you mean this may cause profile inaccurate?

**Comment by stalary** [comment]
It looks like you need rebase~

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
@wangbo please rebase to pass the check

**Review by yiguolei — APPROVED** [review]
lgtm

**Pull request merged**